src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Model/ServerKeyType.cs

```diff
@@ -0,0 +1,31 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model
+{
+    /// <summary>
+    /// Enum representing the allowed managed instance Key types
+    /// </summary>
+    public enum ServerKeyType
+    {
+        AzureKeyVault,
```

Comment:
Why is this different than EncryptionProtectorType?

Reply:
Key and EncryptionProtector are logically different objects hence their types are different enums.